### ISSUE TITLE
Limit

### ISSUE THREAD

**Issue by goocastro** [ankane/groupdate#117]
Using `limit()` returns unrelated number of days:

`Client.group_by_day(:created_at, format: "%d/%m").limit(2).count` returns 3 days

`Client.group_by_day(:created_at, format: "%d/%m").limit(3).count` returns 9 days

`Client.group_by_day(:created_at, format: "%d/%m").limit(5).count` returns 12 days

`Client.group_by_day(:created_at, format: "%d/%m").limit(6).count` returns 17 days

**Comment by goocastro** [comment]
Ended up using `range: 2.weeks.ago.midnight..Time.now`, and it's working fine for me.

Is it the preferred way? `limit` isn't supposed to work anyway?

**Comment by ankane** [comment]
I haven't tried limit before, but the odd behavior makes since from the implementation.  What behavior were you expecting?  I think that's the most important question to determine how we should handle it.  One option would be to throw an error, but could be a better one.

**Comment by goocastro** [comment]
Hi Andrew!

Since I am grouping by day, I expected `limit(n)` to return `n` number of days. But as I said earlier `range:` can also yield the desired result.

But I think `limit` could be a nice shorthand for that purpose.

Regards!

**Comment by ankane** [comment]
Makes sense, we could use `limit` like the `:last` option.  Happy to accept a PR for it.
